sw/device/tests/dif/dif_rv_timer_sanitytest.c

```diff
@@ -77,19 +77,29 @@ bool test_main(void) {
   uint64_t current_time;
   CHECK(dif_rv_timer_counter_read(&timer, kHart, &current_time) ==
         kDifRvTimerOk);
+
+  uint64_t timer_threshold = current_time + kDeadline;
   LOG_INFO("Current time: %d; timer theshold: %d", (uint32_t)current_time,
-           (uint32_t)(current_time + kDeadline));
+           (uint32_t)(timer_threshold));
   CHECK(dif_rv_timer_arm(&timer, kHart, kComparator,
                          current_time + kDeadline) == kDifRvTimerOk);
 
   irq_fired = false;
   CHECK(dif_rv_timer_counter_set_enabled(&timer, kHart, kDifRvTimerEnabled) ==
         kDifRvTimerOk);
 
-  while (!irq_fired) {
-    LOG_INFO("Waiting...");
-    wait_for_interrupt();
-  }
+  LOG_INFO("Waiting...");
+  wait_for_interrupt();
+  CHECK(irq_fired);
+
+  CHECK(dif_rv_timer_counter_read(&timer, kHart, &current_time) ==
```

Comment:
Declare a new variable instead of re-using an old one, for better declaration-site clarity. The compiler will likely merge them to the same stack location anyway.